apps/verticalhome/js/pin_app_navigation.js

```diff
@@ -7,37 +7,61 @@
    * This is PinAppNavigation's constructor.
    * @param {Object} list [DOM container that contains pin elements]
    */
-  function PinAppNavigation(list) {
-    if (!list && typeof list !== 'object') {
-      console.error('List of pin apps must be an object');
-      return;
+  function PinAppNavigation(list, selector) {
+    if (!list) {
+      console.error('List must be a dom container with navigated elements');
```

Comment:
let's throw error in case if incomplete/wrong arguments were passed?
e.g.: ``throw new Error('list must be a DOM container with elements to be navigated');``